streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -61,6 +61,7 @@ import org.apache.spark.util.{CallSite, ShutdownHookManager, ThreadUtils, Utils}
  * `context.awaitTermination()` allows the current thread to wait for the termination
  * of the context by `stop()` or by an exception.
  */
+@deprecated
```

Comment:
We should probably leave a short note to promote moving to Structured Streaming.

> This is deprecated as of Spark 3.4.0. There are no longer updates to DStream and it’s a legacy project. There is a newer and easier to use streaming engine in Spark called Structured Streaming. You should use Spark Structured Streaming for your streaming applications.

Reply:
Leave a note in this file or other public document?

Reply:
https://www.scala-lang.org/api/2.12.5/scala/deprecated.html
You can give a note on the annotation.

`@deprecated("this method will be removed", "FooLib 12.0")`

Let's have a direct short message in this note, and in the classdoc let's find a tag for deprecation (if there's no such thing then just start with `NOTE: `) and have a full description. For version, Spark 3.4.0.

Reply:
Use `@deprecated` in Scala classdoc to provide a full note.

https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/expressions/udaf.scala#L25-L35

Reply:
Furthermore, I just pulled the code and opened in IDEA, and looked through the test suite. companion object needs its own deprecation annotation. It makes sense because the class of companion object is different, having `$` as postfix. Could you please do the same for StreamingContext companion object as well?